deepspeed/module_inject/containers/llama.py

```diff
@@ -122,10 +122,16 @@ def __init__(self, client_module, inference=True):
             LLAMALayerPolicy._orig_layer_class = None
 
     def get_hidden_heads(self):
-        return self.client_module.self_attn.q_proj.weight.shape[1], \
-                self.client_module.self_attn.num_heads, \
-                self.client_module.input_layernorm.variance_epsilon, \
-                self.client_module.mlp.gate_proj.weight.shape[0]
+        try: # for zero stage 3
+            return self.client_module.self_attn.q_proj.weight.ds_shape[1], \
+            self.client_module.self_attn.num_heads, \
+            self.client_module.input_layernorm.variance_epsilon, \
+            self.client_module.mlp.gate_proj.weight.ds_shape[0]
+        except:
```

Comment:
Let's avoid using a base `except` block. We can refactor this to something like:
```python
hidden_heads = (
    getattr(self.client_module.self_attn.q_proj.weight, "ds_shape", self.client_module.self_attn.q_proj.weight.shape)[1],
    self.client_module.self_attn.num_heads,
    self.client_module.input_layernorm.variance_epsilon,
    getattr(self.client_module.mlp.gate_proj.weight, "ds_shape", self.client_module.mlp.gate_proj.weight.shape)[0],
)
return hidden_heads
```

Reply:
Or we could add the exception we expect:
`except AttributeError`

Reply:
Thank you, Michael! That was a good suggestion. I have refactored the code as suggested and also added a few more fixes to bugs in llama support.